play-ahc-ws-standalone/src/test/scala/play/libs/ws/ahc/AhcWSRequestSpec.scala

```diff
@@ -156,19 +167,19 @@ class AhcWSRequestSpec extends Specification with Mockito with DefaultBodyReadab
       val client = mock[StandaloneAhcWSClient]
       val request = new StandaloneAhcWSRequest(client, "http://example.com", /*materializer*/ null)
       request.setBody(body("HELLO WORLD"))
-      request.addHeader("Content-Type", "application/json")
-      request.addHeader("Content-Type", "application/xml")
+      request.addHeader("Content-Type", "application/json") // will be ignored since body already sets content type
       val req = request.buildRequest()
-      req.getHeaders.get("Content-Type") must be_==("application/json")
+      req.getHeaders.get("Content-Type") must be_==("text/plain")
     }
 
     "only send first content type header and keep the charset if it has been set manually with a charset" in {
       val client = mock[StandaloneAhcWSClient]
       val request = new StandaloneAhcWSRequest(client, "http://example.com", /*materializer*/ null)
-      request.setBody(body("HELLO WORLD"))
       request.addHeader("Content-Type", "application/json; charset=US-ASCII")
       request.addHeader("Content-Type", "application/xml")
+      request.setBody(body("HELLO WORLD")) // content type is not overwritten here as its already set before
       val req = request.buildRequest()
+      println("Headers:" + req.getHeaders)
```

Comment:
Remove this. :-)

Reply:
damnit, forgot to remove it, used for debugging :)